goalc/compiler/compilation/Math.cpp

```diff
@@ -14,6 +14,10 @@ MathMode Compiler::get_math_mode(const TypeSpec& ts) {
     return MATH_FLOAT;
   }
 
+  if (m_ts.tc(m_ts.make_typespec("pointer"), ts)) {
+    return MATH_POINTER;
```

Comment:
I don't think we want to add another math mode.  If we did, we'd need to update all the other math forms to throw an appropriate error if they got a pointer (right now, they would `assert(false)`) The idea behind math modes was that the "mode" of a math operation is determined by the type of the first argument, and then all the rest of the arguments are converted to the same mode with `to_math_type`. 

This doesn't really apply to `logand` after this change so it might be easier to do something like:
`if (is_integer(first->type()) ` and similar.